lib/entities/game.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+class Game
+  attr_reader :difficulty, :total_attempts, :used_attempts, :total_hints, :used_hints, :secret_code
+
+  SECRET_CODE_LENGTH = 4
+  ELEMENT_VALUE_RANGE = (1..6).freeze
+
+  DELIMITER = ''
+  EXACT_MATCH = '+'
+  NUMBER_MATCH = '-'
+
+  def initialize(difficulty)
+    @difficulty = difficulty[:level]
+    @total_attempts = difficulty[:attempts]
+    @used_attempts = 0
+    @total_hints = difficulty[:hints]
+    @used_hints = 0
+    @secret_code = generate
+    @shuffled_code = @secret_code.shuffle
+  end
+
+  def use_hint
+    increment_used_hints
+    @shuffled_code.shift
+  end
+
+  def hints_available?
+    @used_hints >= @total_hints
+  end
+
+  def win?(guess_code)
+    convert_to_digit_array(guess_code) == @secret_code
+  end
+
+  def loss?
+    @used_attempts == @total_attempts
+  end
+
+  def increment_used_attempts
+    @used_attempts += 1
+  end
+
+  def mark_guess(guess_code)
+    @converted_input = convert_to_digit_array(guess_code)
+    cloned_code = @secret_code.clone
+    marked_guess = []
```

Comment:
remove this variable and try to use something like ```map``` in your ```_match``` methods for simplifying